Compilation error related to thorvg and webp with webp module disabled

### Tested versions

4.3 bdc0316217940a8ccc80ce536547d42e6477adf4

### System information

Ubuntu 22.04 CI

### Issue description

Cannot compile Godot with command
```
scons -j2 dev_build=yes target=editor use_asan=yes use_ubsan=yes linker=gold module_basis_universal_enabled=no module_bmp_enabled=no module_camera_enabled=no module_csg_enabled=no module_cvtt_enabled=no module_dds_enabled=no module_denoise_enabled=no module_enet_enabled=no module_etcpak_enabled=no brotli=no module_gltf_enabled=no module_gridmap_enabled=no module_hdr_enabled=no module_jpg_enabled=no module_lightmapper_rd_enabled=no module_mbedtls_enabled=no module_meshoptimizer_enabled=no module_minimp3_enabled=no module_mobile_vr_enabled=no module_msdfgen_enabled=no module_multiplayer_enabled=no module_noise_enabled=no module_ogg_enabled=no module_openxr_enabled=no module_raycast_enabled=no module_regex_enabled=no module_squish_enabled=no graphite=no module_text_server_adv_enabled=no module_tga_enabled=no module_theora_enabled=no module_tinyexr_enabled=no module_upnp_enabled=no module_vhacd_enabled=no module_vorbis_enabled=no module_webp_enabled=no module_webrtc_enabled=no module_websocket_enabled=no module_webxr_enabled=no module_xatlas_unwrap_enabled=no
```

```
 Building compilation database compile_commands.json
progress_finish(["progress_finish"], [])
Generating servers/rendering/renderer_rd/shaders/environment/voxel_gi_sdf.glsl.gen.h ...
Generating servers/rendering/renderer_rd/shaders/giprobe_write.glsl.gen.h ...
Linking Static Library core/libcore.linuxbsd.editor.dev.x86_64.san.a ...
Ranlib Library core/libcore.linuxbsd.editor.dev.x86_64.san.a ...
Linking Program bin/godot.linuxbsd.editor.dev.x86_64.san ...
thirdparty/thorvg/src/loaders/external_webp/tvgWebpLoader.cpp:37: error: undefined reference to 'WebPDecodeBGRA'
thirdparty/thorvg/src/loaders/external_webp/tvgWebpLoader.cpp:65: error: undefined reference to 'WebPFree'
thirdparty/thorvg/src/loaders/external_webp/tvgWebpLoader.cpp:89: error: undefined reference to 'WebPGetInfo'
thirdparty/thorvg/src/loaders/external_webp/tvgWebpLoader.cpp:115: error: undefined reference to 'WebPGetInfo'
collect2: error: ld returned 1 exit status
scons: *** [bin/godot.linuxbsd.editor.dev.x86_64.san] Error 1
scons: building terminated because of errors.
```

### Steps to reproduce

Above

### Minimal reproduction project (MRP)

Above

--- a/modules/svg/SCsub
+++ b/modules/svg/SCsub
@@ -26,7 +26,6 @@ thirdparty_sources = [
     "src/loaders/raw/tvgRawLoader.cpp",
     # image loaders
     "src/loaders/external_png/tvgPngLoader.cpp",
-    "src/loaders/external_webp/tvgWebpLoader.cpp",
     "src/loaders/jpg/tvgJpgd.cpp",
     "src/loaders/jpg/tvgJpgLoader.cpp",
     # renderer common
@@ -59,6 +58,10 @@ thirdparty_sources = [
     "src/renderer/sw_engine/tvgSwStroke.cpp",
 ]
 
+if env["module_webp_enabled"]:
+    thirdparty_sources += ["src/loaders/external_webp/tvgWebpLoader.cpp"]
+    env_svg.Append(CPPDEFINES=["THORVG_WEBP_LOADER_SUPPORT"])
+
 thirdparty_sources = [thirdparty_dir + file for file in thirdparty_sources]
 
 env_svg.Prepend(CPPPATH=[thirdparty_dir + "inc"])
@@ -76,12 +79,15 @@ env_thirdparty.Prepend(
         thirdparty_dir + "src/renderer/sw_engine",
         thirdparty_dir + "src/loaders/raw",
         thirdparty_dir + "src/loaders/external_png",
-        thirdparty_dir + "src/loaders/external_webp",
         thirdparty_dir + "src/loaders/jpg",
-        "#thirdparty/libpng",
-        "#thirdparty/libwebp/src",
     ]
 )
+if env["builtin_libpng"]:
+    env_thirdparty.Prepend(CPPPATH=["#thirdparty/libpng"])
+if env["module_webp_enabled"]:
+    env_thirdparty.Prepend(CPPPATH=[thirdparty_dir + "src/loaders/external_webp"])
+    if env["builtin_libwebp"]:
+        env_thirdparty.Prepend(CPPPATH=["#thirdparty/libwebp/src"])
 
 env_thirdparty.add_source_files(thirdparty_obj, thirdparty_sources)
 env.modules_sources += thirdparty_obj

--- a/thirdparty/thorvg/inc/config.h
+++ b/thirdparty/thorvg/inc/config.h
@@ -5,8 +5,10 @@
 #define THORVG_SVG_LOADER_SUPPORT
 #define THORVG_PNG_LOADER_SUPPORT
 #define THORVG_JPG_LOADER_SUPPORT
-#define THORVG_WEBP_LOADER_SUPPORT
 #define THORVG_THREAD_SUPPORT
+
+// Added conditionally if webp module is enabled.
+//#define THORVG_WEBP_LOADER_SUPPORT
 
 // For internal debugging:
 //#define THORVG_LOG_ENABLED

--- a/thirdparty/thorvg/update-thorvg.sh
+++ b/thirdparty/thorvg/update-thorvg.sh
@@ -38,8 +38,10 @@ cat << EOF > ../inc/config.h
 #define THORVG_SVG_LOADER_SUPPORT
 #define THORVG_PNG_LOADER_SUPPORT
 #define THORVG_JPG_LOADER_SUPPORT
-#define THORVG_WEBP_LOADER_SUPPORT
 #define THORVG_THREAD_SUPPORT
+
+// Added conditionally if webp module is enabled.
+//#define THORVG_WEBP_LOADER_SUPPORT
 
 // For internal debugging:
 //#define THORVG_LOG_ENABLED

